### PR TITLE
Refactor backend code / switch to new app structure

### PR DESCRIPTION
**Pull request by splitt3r**
I wanted to let you know / review the refactoring i think otherwise this will lead to heavy merge conflicts.

### CODE REVIEW

**Comment by dartcafe** [comment]
Nevertheless: Great work.

File: lib/Db/CommentMapper.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+use OCP\IDBConnection;
+
+class CommentMapper extends Mapper
+{
+
+    /**
+     * CommentMapper constructor.
+     * @param IDBConnection $db
+     */
+    public function __construct(IDBConnection $db)
```
Comment (dartcafe) [inline lib/Db/CommentMapper.php]: Is that intended?
Shouldn't it be `public function __construct(IDBConnection $db) {`
- Opening braces of blocks are on the same line as the definition
Source: https://docs.nextcloud.com/server/12/developer_manual/general/codingguidelines.html

Comment (splitt3r) [reply lib/Db/CommentMapper.php]: Good catch. I will update that.

**Comment by dartcafe** [comment]
Oops. Was too fast. Still WIP...

**Comment by dartcafe** [comment]
Is it still WIP?

**Comment by dartcafe** [comment]
OK, I signed in and activated travis CI. But I cannot activate the project for travis CI. @v1r0x can you edit rights for the project, or is it managed by the nextcloud organisation. 

Travis says: `You don't have sufficient rights to enable this repo on Travis.
Please contact the admin to enable it or to receive admin rights yourself.`

**Comment by splitt3r** [comment]
I will do some final tests tomorrow and will create the unit tests and other refactoring tasks in the next release.

**Comment by xXSTrikeXx** [comment]
How to get the repository cloned to my server? Wanna test it to, to force the merge 🗡  :D

**Comment by splitt3r** [comment]
@xXSTrikeXx every help is welcome 👍 https://help.github.com/articles/checking-out-pull-requests-locally/ should help

**Comment by xXSTrikeXx** [comment]
Like so: "git fetch original splitt3r:feature/refactor"?

**Comment by splitt3r** [comment]
@xXSTrikeXx good question ^^ I think it´s way more simple to just check out my branch with `git clone -b feature/refactor https://github.com/splitt3r/polls.git`

@dartcafe @v1r0x it´s ready for testing now :)

**Comment by splitt3r** [comment]
@dartcafe the new Layout rocks 💓

**Comment by dartcafe** [comment]
@xXSTrikeXx This is because, we don't check the state of the row. If all options are on maybe, the button doesn't recognize that and offers to change the state to maybe for all options. Nice find... :-) I will think it over, if we can optimize that.

**Comment by dartcafe** [comment]
@splitt3r Semms to meet the taste of some people, yes. Thanks.

**Comment by splitt3r** [comment]
@xXSTrikeXx @dartcafe how about creating an issue? I thnik the js rework is on our ToDo list so we could consider investigating while rewriting it.

**Comment by xXSTrikeXx** [comment]
@splitt3r @dartcafe 
I found it out while write it, that all works fine. But I have already written it. Just wanted to mention it :)

**Comment by splitt3r** [comment]
@v1r0x lets merge this and close #166 so we can do the final tests and fixes in the RC-0.8 branch.

**Pull request merged**